File: java/program/Lokomotif.java
public class Lokomotif extends SaranaKereta {
    // Attributes
    private Mesin jenisMesin;
    private String konfigurasiGandar;
    private String jenisMotorTraksi;

    // Default Constructor
    public Lokomotif(){
        super();
        this.jenisMesin = new Mesin();
        this.konfigurasiGandar = "";
        this.jenisMotorTraksi = "";
    }

    // Parameterized Constructor
    public Lokomotif(String kodeSarana, String namaSarana, int beratSarana, String tipeBogie, Mesin jenisMesin, String konfigurasiGandar, String jenisMotorTraksi) {
        super(kodeSarana, namaSarana, beratSarana, tipeBogie);
        this.jenisMesin = jenisMesin;
        this.konfigurasiGandar = konfigurasiGandar;
        this.jenisMotorTraksi = jenisMotorTraksi;
    }

    // Method to print info
    public void printInfo() {
        System.out.println("\n=== Info Lokomotif (" + this.getKodeSarana() + ") " + this.getNamaSarana() + " ===");
        System.out.println("Berat Sarana: " + this.getBeratSarana() + " kg");
        System.out.println("Tipe Bogie: " + this.getTipeBogie());
        System.out.println("Jenis Mesin: " + this.jenisMesin.getTipeMesin() + ", Daya: " + this.jenisMesin.getDayaMesin() + " HP, Silinder: " + this.jenisMesin.getSilinder());
        System.out.println("Konfigurasi Gandar: " + this.konfigurasiGandar);
        System.out.println("Jenis Motor Traksi: " + this.jenisMotorTraksi);
    }

    // Getter and Setter for jenisMesin
    public Mesin getJenisMesin() {return this.jenisMesin;}
    public void setJenisMesin(Mesin mesin) {this.jenisMesin = mesin;}

    // Getter and Setter for konfigurasiGandar
    public String getKonfigurasiGandar() {return this.konfigurasiGandar;}
    public void setKonfigurasiGandar(String konfigurasi) {this.konfigurasiGandar = konfigurasi;}
    
    // Getter and Setter for jenisMotorTraksi
    public String getJenisMotorTraksi() {return this.jenisMotorTraksi;}
    public void setJenisMotorTraksi(String jenis) {this.jenisMotorTraksi = jenis;}
}

File: cpp/program/Lokomotif.cpp
#pragma once
#include <iostream>
#include "Mesin.cpp"
#include "SaranaKereta.cpp"

// Class Lokomotif
class Lokomotif : public SaranaKereta
{
// Attributes
private:
    Mesin jenisMesin;
    string konfigurasiGandar;
    string jenisMotorTraksi;

public:
    // Default Constructor
    Lokomotif() : SaranaKereta() {
        jenisMesin = Mesin();
        konfigurasiGandar = "";
        jenisMotorTraksi = "";
    }

    // Parameterized Constructor
    Lokomotif(string kodeSarana, string namaSarana, int beratSarana, string tipeBogie, Mesin jenisMesin, string konfigurasiGandar, string jenisMotorTraksi): SaranaKereta(kodeSarana, namaSarana, beratSarana, tipeBogie) {
        this->jenisMesin = jenisMesin;
        this->konfigurasiGandar = konfigurasiGandar;
        this->jenisMotorTraksi = jenisMotorTraksi;
    }

    void printInfo() {
        cout << endl;
        cout << "=== Info Lokomotif " << "(" << getKodeSarana() << ") " << getNamaSarana() << " ===" << endl;
        cout << "Berat Sarana: " << getBeratSarana() << " kg" << endl;
        cout << "Tipe Bogie: " << getTipeBogie() << endl;
        cout << "Jenis Mesin: " << jenisMesin.getNamaMesin() << ", Daya: " << jenisMesin.getDayaMesin() << " HP, Silinder: " << jenisMesin.getSilinder() << endl;
        cout << "Konfigurasi Gandar: " << konfigurasiGandar << endl;
        cout << "Jenis Motor Traksi: " << jenisMotorTraksi << endl;
    }

    // Getter and Setter for jenisMesin
    Mesin getJenisMesin() const {return this->jenisMesin;}
    void setJenisMesin(Mesin mesin) {this->jenisMesin = mesin;}

    // Getter and Setter for konfigurasiGandar
    string getKonfigurasiGandar() const {return this->konfigurasiGandar;}
    void setKonfigurasiGandar(string konfigurasi) {this->konfigurasiGandar = konfigurasi;}

    // Getter and Setter for jenisMotorTraksi
    string getJenisMotorTraksi() const {return this->jenisMotorTraksi;}
    void setJenisMotorTraksi(string jenisMotor) {this->jenisMotorTraksi = jenisMotor;}

    // Destructor
    ~Lokomotif(){}
};